src/OpenTelemetry.Instrumentation.StackExchangeRedis/TracerProviderBuilderExtensions.cs

```diff
@@ -53,7 +53,40 @@ public static class TracerProviderBuilderExtensions
     {
         Guard.ThrowIfNull(connection);
 
-        return AddRedisInstrumentation(builder, name: null, connection, configure: null);
+        return AddRedisInstrumentation(builder, name: null, connection, configure: null, serviceKey: null);
+    }
+
+    /// <summary>
+    /// Enables automatic data collection of outgoing requests to Redis.
+    /// </summary>
+    /// <param name="builder"><see cref="TracerProviderBuilder"/> being configured.</param>
+    /// <param name="serviceKey">Optional service key used to retrieve the <see cref="IConnectionMultiplexer"/> to instrument from the <see cref="IServiceProvider" />.</param>
+    /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+    public static TracerProviderBuilder AddRedisInstrumentation(
+        this TracerProviderBuilder builder,
+        object serviceKey)
+    {
+        Guard.ThrowIfNull(serviceKey);
+
+        return AddRedisInstrumentation(builder, name: null, connection: null, configure: null, serviceKey);
+    }
+
+    /// <summary>
+    /// Enables automatic data collection of outgoing requests to Redis.
+    /// </summary>
+    /// <param name="builder"><see cref="TracerProviderBuilder"/> being configured.</param>
+    /// <param name="serviceKey">Optional service key used to retrieve the <see cref="IConnectionMultiplexer"/> to instrument from the <see cref="IServiceProvider" />.</param>
+    /// <param name="configure">Callback to configure options.</param>
+    /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+    public static TracerProviderBuilder AddRedisInstrumentation(
+        this TracerProviderBuilder builder,
+        object serviceKey,
+        Action<StackExchangeRedisInstrumentationOptions> configure)
```

Comment:
Do we need a way to specify the `name`? As it stands, if I use this API the `configure` callback will configure the `StackExchangeRedisInstrumentationOptions` for all connections. What if I only want to configure the options for this specific connection?